packages/block-editor/src/components/image-size-control/index.js

```diff
@@ -70,47 +111,28 @@ export default function ImageSizeControl( {
 							size="__unstable-large"
 						/>
 					</HStack>
-					<HStack>
-						<ButtonGroup aria-label={ __( 'Image size presets' ) }>
-							{ IMAGE_SIZE_PRESETS.map( ( scale ) => {
-								const scaledWidth = Math.round(
-									imageWidth * ( scale / 100 )
-								);
-								const scaledHeight = Math.round(
-									imageHeight * ( scale / 100 )
-								);
-
-								const isCurrent =
-									currentWidth === scaledWidth &&
-									currentHeight === scaledHeight;
-
-								return (
-									<Button
-										key={ scale }
-										size="small"
-										variant={
-											isCurrent ? 'primary' : undefined
-										}
-										isPressed={ isCurrent }
-										onClick={ () =>
-											updateDimensions(
-												scaledHeight,
-												scaledWidth
-											)
-										}
-									>
-										{ scale }%
-									</Button>
-								);
-							} ) }
-						</ButtonGroup>
-						<Button
-							size="small"
-							onClick={ () => updateDimensions() }
-						>
-							{ __( 'Reset' ) }
-						</Button>
-					</HStack>
+					<ToggleGroupControl
+						label={ __( 'Image size presets' ) }
+						hideLabelFromVision
+						onChange={ ( scale ) =>
+							handleUpdateDimensions( scale )
+						}
+						value={ addSelectedScaleValue() }
+						isBlock
+						isDeselectable
```

Comment:
As discussed here, `isDeselectable` doesn't really work with text options.

One option could be to add an "Auto" option which replaces the 'reset' button. What do you think, @mirka @t-hamano ?

Reply:
If we remove the `isDeselectable` prop, keyboard users won't be able to set a custom value:

https://github.com/user-attachments/assets/7d9074e3-c101-4b08-97d5-b52176a8f01b

As I commented [here](https://github.com/WordPress/gutenberg/pull/65340#issuecomment-2360428325) too, we might need to add an option called "Custom".

Reply:
> keyboard users won't be able to set a custom value

This isn't inherently true — it's only because the logic of `addSelectedScaleValue` doesn't account for it. In fact the current logic sets invalid values on the ToggleGroupControl, when it should be set back to `undefined` instead. (cc @hbhalodia)

For the sake of this PR, I'm ok with just replicating the existing functionality. And as it turns out, the Reset button doesn't actually reset, it just sets it to 100%, which is different from an actual reset. So to actually reset, the user needs to clear the width/height input fields. I think that means we don't even need a Reset button nor a "Custom" option to replicate the existing functionality.

Reply:
Hi @mirka, If I had read it correct, need to update the logic on `addSelectedScaleValue()` function should return `undefined`, when we set custom values? is that something need to changed.

Thank You,

Reply:
> it's only because the logic of addSelectedScaleValue doesn't account for it. In fact the current logic sets invalid values on the ToggleGroupControl, when it should be set back to undefined instead.

Hi @mirka, I tried to set directly undefined on that logic, seems like anyhow keyboard user would ultimately land on the first option of the `ToggleGroupControl`, which ultimately sets the first option even we have undefined set on that logic. 

Can you please let me know how to set it undefined for keyboard users on that logic, because ultimately when user lands on the `toggleGroupControl` it would call `handleUpdateDimension()` which ultimately sets the dimension based on the scale size, and it would remove the custom dimension set.

Reply:
@hbhalodia Yes, thanks for the fix! The problem you encountered with keyboard users is a bug in ToggleGroupControl (#62981). We'll try to get that fixed as soon as possible, so let's put this PR on hold for a bit and I'll report back.

Reply:
@hbhalodia The bug fix has landed 🎉 Could you pull in the changes in latest trunk and try again please?

Reply:
Great, thanks @mirka, I would try this and add the changes.

Reply:
Hi @mirka, I tried to update the logic for `addSelectedScaleValue()` function to return undefined, if the scaleSize is not in image presets. However, I need some help in checking the logic is correct or not because I am facing some issue, When user adds some custom width and height and does not select the `ToggleGroupControl` values it is working as expected, however when user chooses any of values from option and then tries to set the customValues it is working if user directly saves, else if user choose custom values and tabs again then `ToggleGroupControl` values is selected, not sure what I am missing here. Below is the video of demonstration.


https://github.com/user-attachments/assets/fd055d7b-2801-4a0a-adf2-34063390a251

Can you please help updating the logic for setting the initial values on `ToggleGroupControl` and onChange as well and if logic in `addSelectedValue()` is accurate or not.

Thank You 🙇.

Reply:
After investigating this issue, I found that when the `ToggleGroupControl` component is focused, `onChange` fires and the value that actually exists in the options is referenced. This may be one of the causes of the unintended behavior.

I thought that the side effects of the `useDimensionHandler` hook might be affecting it, but deleting [useEffect](https://github.com/WordPress/gutenberg/blob/98f3f6a5b3cccfcc8d94cefa9a4bee9750478132/packages/block-editor/src/components/image-size-control/use-dimension-handler.js#L20-L47) did not solve the problem. Therefore, I still don't know what is causing the unintended onChange event 🤔

Please refer to the verification video below:

https://github.com/user-attachments/assets/d087cae1-5a4b-437a-bdd9-f8d78bfe8d03

Reply:
I looked into this, and I think it's a bug in ToggleGroupControl 😭 Sorry. Just a moment while I address this at the root.

Reply:
Fix proposed in #66151

Reply:
Thanks, @mirka for getting the bug fixed.

Reply:
Fix is merged. Thank you for your patience!

Reply:
Hi @mirka @t-hamano, This is now looking good. Below is the demo screencast. I guess we can go ahead with the PR, let me know if I need to address any other feedbacks.

https://github.com/user-attachments/assets/e60a9ae6-e629-4401-967f-d7a49af7cb4f

Thank You,

Reply:
I may have lost the plot a little bit, but I noticed that unit tests are failing because they require a "reset" button. With the changes from this PR, would a user be able to reset the value at all? Is that something that we should retain?

Again, apologies in case I missed the part of the conversation where we reached an agreement on that aspect.

Reply:
> would a user be able to reset the value at all? Is that something that we should retain?

The reset button in the original was the functional equivalent of re-selecting the "100%" option. The user can also still delete the values from the number inputs. So, I'd say no functionality is lost.

Reply:
Thank you, Lena and Aki — much more clear now. We established that no functionality is lost.

The lack of consistency is also bothering me. I say we merge this PR with the current UX, and then we look at opportunities for improving consistency across other PRs